inc/saivlan.h

```diff
@@ -484,6 +484,193 @@ typedef enum _sai_vlan_stat_t
 
 } sai_vlan_stat_t;
 
+/**
+ * @brief VLAN Stack stage
+ */
+typedef enum _sai_vlan_stack_stage_t
+{
+    SAI_VLAN_STACK_STAGE_INGRESS,
+
+    SAI_VLAN_STACK_STAGE_EGRESS,
+
+} sai_vlan_stack_stage_t;
+
+/**
+ * @brief VLAN Stack action
+ *
+ * If the action is SAI_VLAN_STACK_ACTION_SWAP, the original vlan id of the frame will be replaced
+ * with new vlan id and the new vlan id will be placed in the original tag position.
+ * If the action is SAI_VLAN_STACK_ACTION_PUSH, the original vlan id of the frame will be kept if
+ * exists and the new vlan id will be added as the outer vlan tag.
+ * If the action is SAI_VLAN_STACK_ACTION_POP, the original vlan id of the frame will be removed
+ * if exists.
+ */
+typedef enum _sai_vlan_stack_action_t
+{
+    SAI_VLAN_STACK_ACTION_SWAP,
+
+    SAI_VLAN_STACK_ACTION_PUSH,
+
+    SAI_VLAN_STACK_ACTION_POP,
+
+} sai_vlan_stack_action_t;
+
+/**
+ * @brief VLAN Stack match type
+ */
+typedef enum _sai_vlan_stack_match_type_t
+{
+    SAI_VLAN_STACK_MATCH_TYPE_INNER,
+
+    SAI_VLAN_STACK_MATCH_TYPE_OUTER,
+
```

Comment:
Please enhance this to match multi-tag also. To start with a double-tag matching. 
ex: SAI_VLAN_STACK_MATCH_TYPE_MULTI with stack depth 2?

Reply:
Sorry, I don't fully understand the operation you mentioned.
Can you check if the current design is enough for your desire? Or can you give me some examples about your mentioned operation.